lib/new_relic/agent.rb

```diff
@@ -389,6 +392,92 @@ def record_custom_event(event_type, event_attrs)
       nil
     end
 
+    # Records user feedback events for LLM applications. This API must pass
+    # the current trace id as a parameter, which can be obtained using:
+    #
+    #   NewRelic::Agent::Tracer.current_trace_id
+    #
+    # @param [String] ID of the trace where the chat completion(s) related
+    #   to the feedback occurred.
+    #
+    # @param [String or Integer] Rating provided by an end user
+    #   (ex: “Good", "Bad”, 1, 2, 5, 8, 10).
+    #
+    # @param [optional, String] Category of the feedback as provided by the
+    #   end user (ex: “informative”, “inaccurate”).
+    #
+    # @param start_time [optional, String] Freeform text feedback from an
+    #   end user.
+    #
+    # @param [optional, Hash] Set of key-value pairs to store any other
+    #   desired data to submit with the feedback event.
+    #
+    # @api public
+    #
+    def record_llm_feedback_event(trace_id:,
+      rating:,
+      category: nil,
+      message: nil,
+      metadata: NewRelic::EMPTY_HASH)
+
+      record_api_supportability_metric(:record_llm_feedback_event)
```

Comment:
Perhaps we should hold off on sending the supportability metric until we've made it past the "is DT on?" check that is performed. Otherwise we'll see a metric suggesting use that isn't actually taking place.

Reply:
I think the intention of these api supportability metric is to indicate that the user is calling them, not necessarily that they are successful. There are other api methods where we record the metric before we check whether we actually want to do something or not (`record_custom_event` for example, records the api metric even if custom events are disabled and nothing will happen). So because of that, i think leaving it as is will keep things in line with how we're treating these metrics for other api methods.

Reply:
Excellent. Alignment with existing agent behavior is certainly the goal here with this PR. I'll address any lingering supportability concerns I have with that approach in a different context. Resolved.